plugins/hdfsviewer/src/azkaban/viewer/hdfs/HdfsBrowserServlet.java

```diff
@@ -344,66 +316,62 @@ private void displayDirPage(FileSystem fs, String user,
-      Path path) throws IOException {
+  private void displayFilePage(FileSystem fs, String user, HttpServletRequest req, HttpServletResponse resp,
+      Session session, Path path) {
 
-    Page page =
-        newPage(req, resp, session, "azkaban/viewer/hdfs/velocity/hdfs-file.vm");
-    page.add("allowproxy", allowGroupProxy);
-    page.add("viewerPath", viewerPath);
-    page.add("viewerName", viewerName);
+    Page page = newPage(req, resp, session, "azkaban/viewer/hdfs/velocity/hdfs-file.vm");
 
     List<Path> paths = new ArrayList<Path>();
     List<String> segments = new ArrayList<String>();
     getPathSegments(path, paths, segments);
 
+    page.add("allowproxy", allowGroupProxy);
+    page.add("viewerPath", viewerPath);
+    page.add("viewerName", viewerName);
+
     page.add("paths", paths);
     page.add("segments", segments);
     page.add("user", user);
     page.add("path", path.toString());
     page.add("homedir", getHomeDir(fs));
 
-    boolean hasSchema = false;
-    int viewerId = -1;
-    for (int i = 0; i < viewers.size(); ++i) {
-      HdfsFileViewer viewer = viewers.get(i);
-      Set<Capability> capabilities = EnumSet.noneOf(Capability.class);
-      try {
+    // disable fs rendering as default.
+    // value will be flipped to false in case everything went out smoothly.
+    // Note : we are capturing the generic exception as we want to render the error gracefully to user in
+    //        any cases error happens.
+    page.add("no_fs", "true");
+
+    try {
+      boolean hasSchema = false;
+      int viewerId = -1;
+      for (int i = 0; i < viewers.size(); ++i) {
+        HdfsFileViewer viewer = viewers.get(i);
+        Set<Capability> capabilities = EnumSet.noneOf(Capability.class);
         capabilities = viewer.getCapabilities(fs, path);
-      } catch (AccessControlException e) {
-        page.add("error_message", "Permission denied (" + viewer.getName()
-            + "): " + e.getMessage());
-        page.add("no_fs", "true");
-        page.render();
-        return;
-      }
-      if (capabilities.contains(Capability.READ)) {
-        if (capabilities.contains(Capability.SCHEMA)) {
-          hasSchema = true;
+        if (capabilities.contains(Capability.READ)) {
+          if (capabilities.contains(Capability.SCHEMA)) {
+            hasSchema = true;
+          }
+          viewerId = i;
+          break;
         }
-        viewerId = i;
-        break;
       }
-    }
 
-    page.add("viewerId", viewerId);
-    page.add("hasSchema", hasSchema);
+      page.add("viewerId", viewerId);
+      page.add("hasSchema", hasSchema);
+      page.add("no_fs", "false");
 
-    try {
       FileStatus status = fs.getFileStatus(path);
       page.add("status", status);
-    } catch (AccessControlException e) {
-      page.add("error_message", "Permission denied: " + e.getMessage());
-    } catch (IOException e) {
-      page.add("error_message", "Error: " + e.getMessage());
+
+    } catch (Exception ex) {
```

Comment:
To reply both - 
1. The e.message will indicate clearly the exception is caused by a denied access,  I figured it is not necessary to write an extra string there saying permission denied where the detailed message will indicate the same . 
2. The idea here is to wrap up ALL the exception and represent it gracefully on the page therefore the goal for this change is to capture all the exceptions, so i think it will not be a good idea to specify types then ORing the result, plus , the java version to compile azkaban is still 1.6